lib/model/node.js

```diff
@@ -1,96 +1,87 @@
-/**
- * Creates a new node.
+ *
+ * @class Node
  * @constructor
- * 
- * @param id - The Node's unique identifier.
- * @param inputs - An array containing the Node's Inputs.
- * @param outputs - An array containing the Node's Outputs.
- * @param styles - An array of strings which contain the styles that will be
- *                 applied to the node in the visualization.
- * @param metadata - A standard JS object containing key-value metadata.
  */
-var Node = function(id, inputs, outputs, styles, metadata) {
-  this._id = id || 0;
-  this._inputs = inputs || [];
-  this._outputs = outputs || [];
-  this._styles = styles || [];
-  this._metadata = metadata || {};
-  this._x = 0;
-  this._y = 0;
-  this._pinned = false;
-};
+var Node = createClass({
+  extend: GraphModelBase,
 
-/**
- * @method getId - Getter method for the id property.
- */
-Node.prototype.getId = function() {
-  return this._id;
-};
+  constructor: function(config) {
 
-/**
- * @method getPosition - Getter method for returning an xy-coordinate pair representing the Node's
- *         position in 2D space.
- */
-Node.prototype.getPosition = function() {
-  return{
-    x: this._x,
-    y: this._y
-  };
-};
+    // re-fire events from the ports
+    this.get('inputs').forEach(function(port) {
+      port.addBubbleTarget(this);
+    }.bind(this));
 
-/**
- * @method setPosition - Setter method for setting an xy-coordinate pair representing the Node's
- *         position in 2D space.
- */
-Node.prototype.setPosition = function(x, y) {
-  this._x = x;
-  this._y = y;
-  this._pinned = true;
+    this.get('outputs').forEach(function(port) {
+      port.addBubbleTarget(this);
+    }.bind(this));
+  },
 
-};
+  instance: {
+    setPosition: function(x, y) {
+      this.set('position', {x: x, y: y});
+    }
+  },
 
-/**
-* @method isPinned - Returns whether the node is pinned or not. 
-*/
-Node.prototype.isPinned = function() {
-  return this._pinned;
-};
+  attrs: {
+    inputs: {
+      value: [],
+      validator: validateArrayOfPortsOfType(IOType.INPUT),
+      initOnly: true
+    },
 
-/**
- * @method getInputs - Getter method for the inputs property.
- */
-Node.prototype.getInputs = function() {
-  return this._inputs;
-};
+    outputs: {
+      value: [],
+      validator: validateArrayOfPortsOfType(IOType.OUTPUT),
+      initOnly: true
+    },
 
-/**
- * @method getOutputs - Getter method for the outputs property.
- */
-Node.prototype.getOutputs = function() {
-  return this._outputs;
-};
+    styles: {
+      //dunno
+    },
 
-/**
- * @method getMeta - Getter method for the metadata property.
- */
-Node.prototype.getMeta = function() {
-  return this._metadata;
-};
+    metadata: {
+      validator: Lang.isObject
+    },
 
-/**
- * @method getStyles - Getter method for the styles property.
- */
-Node.prototype.getStyles = function() {
-  return this._styles;
-};
+    position: {
+      validator: function(val) {
+        return 'x' in val && 'y' in val;
+      }
+    }
+  }
```

Comment:
Yes, it seems to be an oversight in the original data model. But you've got this under control on another branch.